naga/src/front/wgsl/lower/mod.rs

```diff
@@ -2060,11 +2060,9 @@ impl<'source, 'temp> Lowerer<'source, 'temp> {
 
                         lowered_base.map(|base| crate::Expression::AccessIndex { base, index })
                     }
-                    crate::TypeInner::Vector { .. } | crate::TypeInner::Matrix { .. } => {
+                    crate::TypeInner::Vector { .. } => {
                         match Components::new(field.name, field.span)? {
                             Components::Swizzle { size, pattern } => {
-                                // Swizzles aren't allowed on matrices, but
-                                // validation will catch that.
```

Comment:
It does seem like we validate this already.

https://github.com/gfx-rs/wgpu/blob/f6f9233295f1bf49ada1da2d103db0ce087b9eda/naga/src/valid/expression.rs#L350-L362

Maybe the validator wasn't run in https://github.com/gfx-rs/wgpu/issues/5496, but this PR looks good either way.